test-apps/nuxt/pages/index.vue

```diff
@@ -1,27 +1,27 @@
 <script lang="ts" setup>
-const { data } = await useFetch("/api/user");
-if (!data.value) throw createError("Failed to fetch data");
-const user = data.value.user;
-if (!user) await navigateTo("/login");
+const { data, error } = await useFetch<{ user: any }>("/api/user");
+const user = computed(() => data?.value?.user);
+if (error.value) throw createError(error.value.message);
+if (!user.value) navigateTo("/login");
 
 const handleSubmit = async (e: Event) => {
 	if (!(e.target instanceof HTMLFormElement)) return;
 	try {
-		const data = await $fetch("/api/logout", {
+		const { error } = await useFetch("/api/logout", {
 			method: "POST"
 		});
-		if (data) {
-			return;
-		}
+		if (error.value) throw error.value;
```

Comment:
```suggestion
		if (data) {
			return;
		}
```

Reply:
/logout either returns null or an error so this can be removed entirely